torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -106,6 +110,22 @@ namespace c10d {
 
 namespace {
 
+using steady_clock_time_point =
+    std::chrono::time_point<std::chrono::steady_clock>;
+
+std::chrono::milliseconds getRemainingTime(
+    steady_clock_time_point startTime,
+    const std::chrono::milliseconds& timeout) {
+  auto elapsedTime = std::chrono::steady_clock::now() - startTime;
+  auto remainingMillis = timeout -
+      std::chrono::duration_cast<std::chrono::milliseconds>(elapsedTime);
+  const int kMinTimeThresh = 1;
```

Comment:
Oh are we returning 1 as the minimum value here since `wait()` assumes a value of 0 as no timeout? If so, I think we could just rename this function to `checkAndGetRemainingTime` and just throw an exception from here that the barrier timed out.

Reply:
Sounds good, I think it might be cleaner to return something like `-1` in this case and let monitoredBarrier() impl take care of raising the error?